pkg/sql/schemachanger/scbuild/builder_state.go

```diff
@@ -851,6 +851,20 @@ func (b *builderState) ResolveIndex(
 func (b *builderState) ResolveIndexByName(
 	tableIndexName *tree.TableIndexName, p scbuildstmt.ResolveParams,
 ) scbuildstmt.ElementResultSet {
+	// If a table name is specified, confirm its not a systme table first.
+	if tableIndexName.Table.ObjectName != "" {
+		desc := b.resolveRelation(tableIndexName.Table.ToUnresolvedObjectName(), p)
+		if desc == nil {
+			return nil
+		}
+		tableDesc := desc.desc.(catalog.TableDescriptor)
+		b.ensureDescriptor(tableDesc.GetParentSchemaID())
+		if catalog.IsSystemDescriptor(tableDesc) {
+			schemaDesc := b.descCache[tableDesc.GetParentSchemaID()].desc
+			panic(catalog.NewMutableAccessToVirtualSchemaError(schemaDesc.(catalog.SchemaDescriptor)))
+		}
+	}
```

Comment:
This may be out of scope for this change but naively I'd expect this check to be covered by our existing privilege checks. What's the story here?

Reply:
The builder never asks for mutable references to these descriptors so there isn't any way to trigger the errors via the normal code path. We could hit them on execution, but I think that's not the right approach for errors like these @postamar .